gemini_instruments/skycam/adclass.py

```diff
@@ -36,34 +37,23 @@ def _tag_site(self):
 
     @astro_data_descriptor
     def instrument(self):
-        hdr = self.TABLE1.meta.get('header')
-        return hdr.get('TELESCOP').strip('"')
+        return self.phu['TELESCOP'].strip('"')
 
     @astro_data_descriptor
     def object(self):
-        hdr = self.TABLE1.meta.get('header')
         return 'ZENITH'
 
     @astro_data_descriptor
     def exposure_time(self):
-        hdr = self.TABLE1.meta.get('header')
-        return hdr.get('EXPTIME')
+        return self.phu['EXPTIME']
 
     @astro_data_descriptor
     def ra(self):
-        hdr = self.TABLE1.meta.get('header')
-        return hdr.get('RA')
+        return self.phu['RA']
 
     @astro_data_descriptor
     def dec(self):
-        hdr = self.TABLE1.meta.get('header')
-        return hdr.get('DEC')
-    
-    @astro_data_descriptor
-    def ut_datetime(self):
-        hdr = self.TABLE1.meta.get('header')
-        return dateutil.parser.parse(hdr.get('DATE-OBS'))
-        
+        return self.phu['DEC']
```

Comment:
Then there is probably no need to redefine those descriptors (ra, dec, exptime, ...) since the parent class is already getting those values from the primary header ?

Reply:
The code in the base class actually looks a little different to this.  For RA and DEC, if I remove these overrides I no longer get values for the skycam files.  If you want a sample file to play with let me know.

Reply:
Yes, out of curiosity ;)

Reply:
I find this surprising, since the ra() descriptor tries to get the location from the WCS of the first extension -- although the _get_wcs_coords() method should probably convert (CRPIX1-1, CRPIX2-1) instead of the middle of the extension -- and then falls back to the phu['RA'] keyword (and dec() does the same), so I would have expected this to work.

Reply:
I was able to remove exposure_time and ut_time and the files still ingest for archive.  I've left RA and DEC since it seems to not like something about the wcs bit in the base class.  I think that's probably fine for these files, but if someone wants to work through that with me let me know.

Reply:
The WCS call whines about "invalid projection" somewhere deep in the internals and skips over the rest by throwing an exception.  It's a ValueError with this notation in it:

`ERROR 5 in wcsset() at line 2279 of file cextern/wcslib/C/wcs.c:
Invalid parameter value.
ERROR 2 in celset() at line 215 of file cextern/wcslib/C/cel.c:
Invalid projection parameters.
ERROR 2 in zpnset() at line 2213 of file cextern/wcslib/C/prj.c:
Invalid parameters for zenithal/azimuthal polynomial projection.
`

Reply:
Yeah, what the heck is "RA---ZPN"? Google tells me it's a "Zenith Polynomial Projection" but it's missing polynomial keywords. Perhaps it means "RA---AZP"? for Zenithal Projection?

AD-with-gWCS handles that and produces something. And, tbh, the descriptors should now be using the wcs attributes, not reconstructing them from the FITS headers, so those descriptors will need to be refactored anyway.